src/charm.py

```diff
@@ -92,6 +93,11 @@ def _mongodb_client_relation_changed(self, event: ops.framework.EventBase) -> No
         if initial != self._stored.reldata["mongodb"]:
             self._configure_workload(event)
 
+    def _assign_external_hostname(self, hostname) -> str:
```

Comment:
I think the name of `_get_external_hostname` would be more consistent with other methods in this charm. Also, you can access `self.config` here too, which would simplify calls to this function by not needing to pass a parameter, so I think that's a better option. Also, Can you include a docstring?